addons/l10n_es_edi_facturae/models/l10n_es_edi_facturae_certificate.py

```diff
@@ -73,7 +72,20 @@ def _sign_xml(self, edi_data, signature_data):
         root = deepcopy(edi_data)
         cert_private, cert_public = self._decode_certificate()
         public_key_numbers = cert_public.public_key().public_numbers()
-        issuer = ', '.join(sorted(element.rfc4514_string() for element in cert_public.issuer.rdns if re.match(r'[A-Z]{1,2}', element.rfc4514_string())))
+
+        def rank_rfc4514_attr_string(rfc4514_string):
+            attr_name = rfc4514_string.split("=", 1)[0]
+            values = {
+                "CN": 0,
+                "OU": 1,
+                "O": 2,
+                "C": 3,
+               }
+            return values.get(attr_name, 10)
+        rfc4514_attr_strings = [
+            element.rfc4514_string() for element in cert_public.issuer.rdns if rank_rfc4514_attr_string(element.rfc4514_string()) < 4
+        ]
+        issuer = ', '.join(sorted(rfc4514_attr_strings, key=rank_rfc4514_attr_string))
```

Comment:
what about just doing: `dict(el.rfc4514_string().split('=') for el in cert_public.issuer.rdns)` and then collecting the values needed to create the string ?

Reply:
Thanks, your version is much simpler.

I think there should always be an '=' in the `rfc4514_string`.
So the `split` should always return lists of length 2.
https://github.com/pyca/cryptography/blob/d3ff595cccb610753c2cf947197748a8b05cdcbe/src/cryptography/x509/name.py#L181-L196
Or should I rather add an additional step to make sure the elements passed to the dict always have length 2?

Reply:
sounds good like that !